ros/lib/utils.py

```diff
@@ -132,3 +135,14 @@ def insert_performance_profiles(session, system_id, fields):
         new_entry = model_class(**fields)
         session.add(new_entry)
         session.flush()
+
+
+def count_per_state(queryset, custom_filters: dict):
+    return queryset.filter_by(**custom_filters).count() if queryset else None
```

Comment:
In case of `else` condition, do we want to return `None` or something like a `-1` ?

Reply:
@apuntamb I think this data would directly go into the generated pdf, and there would be no use cases for sorting.
@PreetiW Thoughts?

Reply:
> In case of `else` condition, do we want to return `None` or something like a `-1` since we are talking about counts here and also for frontend to be able to consume it better ?

Good point @apuntamb!
@saltgen, in this case can't we simply return `0` than `None`?

Reply:
@saltgen @kgaikwad yes this data we directly use in the pdf report and for now I have put a check for conditions count data that if it is null then display `0` on the frontend but I agree with @kgaikwad it will be good if backend can send 0 for count and percentage when the data is not available. Let me know your thoughts and accordingly will make changes in the frontend.

Reply:
@kgaikwad @PreetiW Done, `count` and `percentage` is now `0` and `0.0` respectively in such cases.
Change pushed

Reply:
@kgaikwad @PreetiW Done, `count` and `percentage` is going to be `0` and `0.0` for such cases.
Change pushed